src/tokenizer_test.cc

```diff
@@ -77,6 +78,18 @@ TEST_F(TokenizerTest, ProcessDouble) {
   EXPECT_TRUE(next->IsEOS());
 }
 
+TEST_F(TokenizerTest, ProcessNaN) {
+  Tokenizer t("NaN");
+  auto next = t.NextToken();
+  ASSERT_TRUE(next != nullptr);
+  EXPECT_TRUE(next->IsDouble());
+  EXPECT_TRUE(std::isnan(next->AsDouble()) != 0);
```

Comment:
std::isnan returns a bool, so just EXPECT_TRUE(std::isnan(next->AsDouble()))

Reply:
Ah, you're right. isnan from C returns an integer but std::isnan returns a bool. Thanks!